core/sched.c

```diff
@@ -181,6 +190,12 @@ void sched_switch(uint16_t other_prio)
 
 NORETURN void sched_task_exit(void)
 {
+    /* Prevent optimization of these symbols */
+    max_threads = max_threads;
```

Comment:
@gebart 
See e3a2dcf for how this could look like. This way these symbols should be optimizied away if a platform doesn't have the `KEEP` directive in the linkerscript, so no overhead for other platforms. Haven't tested though, I'm lacking the toolchains for msp430 and avr.